be/src/vec/functions/function_json.cpp

```diff
@@ -936,6 +937,186 @@ class FunctionJsonValid : public IFunction {
     }
 };
 
+class FunctionJsonMergePreserve : public IFunction {
+public:
+    static constexpr auto name = "json_merge_preserve";
+    static FunctionPtr create() { return std::make_shared<FunctionJsonMergePreserve>(); }
+
+    String get_name() const override { return name; }
+
+    size_t get_number_of_arguments() const override { return 0; }
+
+    bool is_variadic() const override { return true; }
+
+    DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
+        return make_nullable(std::make_shared<DataTypeString>());
+    }
+
+    bool use_default_implementation_for_nulls() const override { return false; }
+
+    rapidjson::Value* merge_list(std::vector<rapidjson::Value*> jsons,
+                                 rapidjson::Document::AllocatorType& allocator) {
+        rapidjson::Value* result =
+                static_cast<rapidjson::Value*>(allocator.Malloc(sizeof(rapidjson::Value)));
+
+        (*result).SetArray();
+        for (auto json : jsons) {
+            if (json->IsArray()) {
+                for (auto itr = json->Begin(); itr != json->End(); itr++) {
+                    rapidjson::Value* node = static_cast<rapidjson::Value*>(
+                            allocator.Malloc(sizeof(rapidjson::Value)));
+                    node->CopyFrom(*itr, allocator);
+                    result->PushBack(*node, allocator);
+                }
+            } else {
+                rapidjson::Value* node =
+                        static_cast<rapidjson::Value*>(allocator.Malloc(sizeof(rapidjson::Value)));
+                node->CopyFrom(*json, allocator);
+                result->PushBack(*node, allocator);
+            }
+        }
+        return result;
+    }
+
+    rapidjson::Value* merge_obejcts(std::vector<rapidjson::Value*> objs,
```

Comment:
write some step or description of ur function

Reply:
done